### PR TITLE
[MM-33020] Make app ID validation more strict

### PR DESCRIPTION
**Pull request by hanzei**
### Summary
Given that we use app IDs in the store where key length is limited we should put a stricter limit on the length. Also, 3 characters is now the minimum limit.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33020

### CODE REVIEW

File: apps/app.go [hunk 1]
```diff
@@ -66,11 +66,18 @@ type ListedApp struct {
 	Labels    []model.MarketplaceLabel `json:"labels,omitempty"`
 }
 
-const MaxAppID = 32
+const (
+	MinAppIDLength = 3
+	MaxAppIDLength = 25
```
Comment (levb) [inline apps/app.go:71]: Why 25? And the comment says, up to 20?

**Review by catalintomai — APPROVED** [review]

File: apps/app.go (hunk 1 above)
Comment (hanzei) [reply apps/app.go:71]: I wasn't aware of the fact that the app id is not longer used directly. With that in mind I'm happy to revert back to 32 bytes. What do you think @levb ?

Comment (levb) [reply apps/app.go:71]: @hanzei should work.

**Comment by levb** [comment]
@hanzei @justinegeffen we should update the docs with this

**Comment by hanzei** [comment]
@levb I didn't found any existing docs on the manifest properties. Do we need to write them from scratch?

**Review by levb — APPROVED** [review]

**Comment by levb** [comment]
@hanzei godoc

**Comment by hanzei** [comment]
@levb Is https://github.com/mattermost/mattermost-plugin-apps/pull/123/files#diff-f41d83db23b9875fe2c189732d707ee45e4f9b2187b5418e2d70f3d7cb55cd02R11-R12 enough?

**Comment by levb** [comment]
Ooops, maybe jumped the gun cc @DHaussermann but unit tests seem to cover the changes.

**Pull request merged**